packages/dds/tree/src/util/rangeMap.ts

```diff
@@ -33,20 +38,36 @@ export class RangeMap<T> {
 	 * and the number of consecutive keys with that same value.
 	 */
 	public get(start: number, length: number): RangeQueryResult<T> {
-		for (const entry of this.entries) {
-			if (entry.start > start) {
-				return { value: undefined, length: Math.min(entry.start - start, length) };
+		// We first check for an entry with a key less than or equal to `start`.
+		{
+			const entry = this.tree.getPairOrNextLower([start]);
+			if (entry !== undefined) {
+				const [entryKey] = entry[0];
+				const { value, length: entryLength } = entry[1];
+
+				const entryLastId = entryKey + entryLength - 1;
+				const overlappingLength = Math.min(entryLastId - start + 1, length);
+				if (overlappingLength > 0) {
+					return { value, start, length: overlappingLength };
+				}
 			}
+		}
+
+		{
+			// There is no value associated with `start`.
+			// Now we need to determine how many of the following keys are also undefined.
+			const key = this.tree.nextHigherKey([start]);
+			if (key !== undefined) {
+				const [entryKey] = key;
 
-			const lastRangeKey = entry.start + entry.length - 1;
-			if (lastRangeKey >= start) {
-				const overlapLength = lastRangeKey - start + 1;
-				return { value: entry.value, length: Math.min(overlapLength, length) };
+				const lastQueryId = start + length - 1;
+				if (entryKey <= lastQueryId) {
+					return { value: undefined, start, length: entryKey - start };
```

Comment:
You don't want to return `value: undefined` in this case, right?

Reply:
This function returns the value for `start`, which is `undefined` in this case.

Reply:
Ah right, sorry, I had the wrong contract in mind. I was thinking we were returning the first defined range, but no.